ray-operator/Dockerfile

```diff
@@ -15,11 +15,10 @@ COPY apis/ apis/
 COPY controllers/ controllers/
 
 # Build
+USER root
```

Comment:
What prompted this change? Could it potentially raise any security concerns?

Reply:
This change was prompted by the switch over to `registry.access.redhat.com/ubi8/go-toolset:1.17.12-11`. The default user in the `go-toolset` image does not have the right permissions to be able to build the binary. It doesn't raise any security concerns as we are specifically setting a non-root user at https://github.com/ray-project/kuberay/pull/1213/files#diff-43d3bfc86d147402a0bff47e59e7aadb5e8a7e3154b32accce70f6e20a824b44R24.